iterations.r

```diff
@@ -50,20 +50,30 @@ test_stations_metadata(stations_metadata_df)
 
 source("gql-queries/vol_qry.r")
 
-stations_metadata_df %>% 
+selected_station <- stations_metadata_df %>% 
   filter(latestData > Sys.Date() - days(7)) %>% 
-  sample_n(1) %$% 
-  vol_qry(
-    id = id,
-    from = to_iso8601(latestData, -4),
-    to = to_iso8601(latestData, 0)
-  ) %>% 
-  GQL(., .url = configs$vegvesen_url) %>%
-  transform_volumes() %>% 
-  ggplot(aes(x=from, y=volume)) + 
-  geom_line() + 
-  theme_classic()
-
+  sample_n(1) # Sampling one station from the data frame
+
+name <- selected_station$name # Defining the station to a variable to be able 
+# to add legend in the ggplot
+
+vol_data <- vol_qry(
+  id = selected_station$id,
+  from = to_iso8601(selected_station$latestData, -4),
+  to = to_iso8601(selected_station$latestData, 0)
+)
+
+# Use vol_data defined earlier to make the ggplot
+vol_data %>% 
+  GQL(.url = configs$vegvesen_url) %>%
+  transform_volumes() %>% # Using the transform from data_transformations
+  mutate(name = name) %>% # Adding the name for the station
+  ggplot(aes(x = from, y = volume, color = name)) + 
+  geom_line() + # Adding a line to the plot
+  scale_color_manual(values = "blue", name = "") + # Manually adding the legend
+  theme_classic() + 
+  xlab("Date") + # Making the axis labels cleaner
+  ylab("Volume")
```

Comment:
Nice plot. Usually, a legend is only used if there are multiple categories in a chart. Since we only have one station in the plot I would drop the legend and rather add a dynamic title. Try to Google how you can get that.